cfg_mgmt/btp_service_binding.py

```diff
@@ -0,0 +1,159 @@
+import copy
+import json
+import logging
+import typing
+import requests
+
+import cfg_mgmt
+import cfg_mgmt.model as cmm
+import ci.log
+import ci.util
+import model
+import model.container_registry
+
+
+ci.log.configure_default_logging()
+logger = logging.getLogger(__name__)
+
+
+def _filter_service_bindings(result: dict, binding_name_prefix: str, instance_id: str) -> tuple[list[str], int]:
+    outdated_ids = []
+    next = 1
+    for item in result["items"]:
```

Comment:
rather pass-in sequence of items, instead of a dict w/ expected key `items`?

it might be worth considering to define a dataclass (convenient deserialisation is offered by `dacite`), e.g.:

```
@dataclasses.dataclass
class ServiceBinding:
 name: str
 service_instance_id: str

 
 # you may optionally add methods..
 def idx(self):
   # add code for parsing idx here?
   return <parsed-idx>

bindings = [dacite.from_dict(data_class=ServiceBinding, data=binding_dict) for binding_dict in bindings]

for binding in bindings:
  if binding.name .....
```

Reply:
It's the only place where the results are used, therefore no worth the effort.